src/logics/SubstLoopBreaker.h

```diff
@@ -0,0 +1,192 @@
+//
+// Created by prova on 21.11.19.
+//
+
+#ifndef OPENSMT_SUBSTLOOPBREAKER_H
+#define OPENSMT_SUBSTLOOPBREAKER_H
+#include <stdint.h>
+#include <PTRef.h>
+#include <Sort.h>
+#include <Alloc.h>
+#include <assert.h>
+#include <PtStructs.h>
+#include <Logic.h>
+
+// For breaking substitution loops
+enum class NStatus { inStack, complete, unseen };
+
+template <class T>
+struct Ref {
+    uint32_t x;
+    Ref() : x(INT32_MAX-1) {}
+    Ref(uint32_t i) : x(i) {}
+    inline friend bool operator== (const Ref& a1, const Ref& a2)   { return a1.x == a2.x; }
+    inline friend bool operator!= (const Ref& a1, const Ref& a2)   { return a1.x != a2.x; }
+    inline friend bool operator< (const Ref& a1, const Ref& a2)    { return a1.x > a2.x;  }
```

Comment:
This is most probably not a problem, I am just curious, why is the operator flipped?
Why `a1 < a2` iff `a1.x > a2.x`?

Reply:
Hehe I think it was just for fun :)

Reply:
OK :)
It's the same for PTRef.
I wonder what would happen if we would flip it to the more natural implementation.
But maybe it's better not to try :)